File: OOP Start/PhonebookApp/Core/MVP/PlainTextModel.java
package PhonebookApp.Core.MVP;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.time.LocalDate;
import PhonebookApp.Config;
import PhonebookApp.Core.Infrastructure.Phonebook;
import PhonebookApp.Core.Models.Contact;

public class PlainTextModel extends Model {
    private String path;

    public PlainTextModel(String path) {
        super();
        this.path = path;
    }

    public void load() {
        try {
            File file = new File(path);
            FileReader fr = new FileReader(file);
            BufferedReader reader = new BufferedReader(fr);
            Phonebook currentBook = currentBook();
            String fdate = reader.readLine();
            while (fdate != null) {
                String firstName = reader.readLine();
                String lastName = reader.readLine();
                String phone = reader.readLine();
                String company = reader.readLine();
                String email = reader.readLine();                
                String description = reader.readLine();
                currentBook.add(new Contact(LocalDate.parse(fdate, Config.ofPattern()), firstName, lastName, phone, company, email, description));
                fdate = reader.readLine();
            }

            reader.close();
            fr.close();
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }
    }

    public void save() {
        try (FileWriter writer = new FileWriter(path, false)) {
            Phonebook currentBook = currentBook();
            for (int i = 0; i < currentBook.count(); i++) {
                Contact contact = currentBook.getContact(i);
                writer.append(String.format("%s\n", contact.getDate().format(Config.ofPattern())));
                writer.append(String.format("%s\n", contact.getFirstName()));
                writer.append(String.format("%s\n", contact.getLastName()));
                writer.append(String.format("%s\n", contact.getPhone()));
                writer.append(String.format("%s\n", contact.getCompany()));
                writer.append(String.format("%s\n", contact.getEmail()));
                writer.append(String.format("%s\n", contact.getDescription()));                
            }

            writer.flush();
            writer.close();
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }
    }
}

File: OOP Start/OOP. Lesson 7/Core/MVP/PlainTextModel.py
import datetime

from Config import Config
from Core.MVP.Model import Model
from Core.Models.Contact import Contact

class PlainTextModel(Model):
    def __init__(self, path):
        super().__init__()
        self.path = path

    def load(self):
        try:
            with open(self.path, mode='r') as file:
                currentBook = self.currentBook
                fdate = file.readline().strip()
                while fdate:
                    firstName = file.readline().strip()
                    lastName = file.readline().strip()
                    phone = file.readline().strip()
                    company = file.readline().strip()
                    email = file.readline().strip()
                    description = file.readline().strip()
                    currentBook.add(Contact(datetime.datetime.strptime(fdate, Config.date_format).date(), firstName, lastName, phone, company, email, description))
                    fdate = file.readline().strip()

        except IOError as ex:
            print(ex)

    def save(self):
        try:
            with open(self.path, mode='w') as file:
                currentBook = self.currentBook
                for i in range(currentBook.count()):
                    contact = currentBook.get_contact(i)
                    fdate = contact.get_date().strftime(Config.date_format)
                    firstName = contact.get_first_name()
                    lastName = contact.get_last_name()
                    phone = contact.get_phone()
                    company = contact.get_company()
                    email = contact.get_email()
                    description = contact.get_description()
                    file.write(f'{fdate}\n{firstName}\n{lastName}\n{phone}\n{company}\n{email}\n{description}\n')

        except IOError as ex:
            print(ex)
